EIPS/eip-5748.md

```diff
@@ -0,0 +1,191 @@
+            require(currentAllowance >= amount, "ERC20: insufficient allowance");
+            unchecked {
+                _approve(owner, spender, currentAllowance - amount, currentExpiration - block.timestamp);
+            }
+        }
+    }
+```
+
+The `allowance()`, `increaseAllowance()`, and `decreaseAllowance()` methods have been modified to accommodate the new features. `allowanceExpiration()` has been added to query the live period of the contract.
+
+```solidity
+        function allowanceExpiration(address owner, address spender) public view returns (uint256) {
+        return _expiration[owner][spender];
+    }
+    
+    
+    function increaseAllowance(address spender, uint256 addedValue) public virtual returns (bool) {
+        address owner = _msgSender();
+        _approve(owner, spender, allowance(owner, spender) + addedValue, DEFAULT_EXPIRATION);
+        return true;
+    }
+
+    function increaseAllowance(address spender, uint256 addedValue, uint256 period) public virtual returns (bool) {
+        address owner = _msgSender();
+        _approve(owner, spender, allowance(owner, spender) + addedValue, period);
+        return true;
+    }
+    
+    
+    function decreaseAllowance(address spender, uint256 subtractedValue) public virtual returns (bool) {
+        address owner = _msgSender();
+        uint256 currentAllowance = allowance(owner, spender);
+        require(currentAllowance >= subtractedValue, "ERC20: decreased allowance below zero");
+        unchecked {
+            _approve(owner, spender, currentAllowance - subtractedValue, DEFAULT_EXPIRATION);
+        }
+
+        return true;
+    }
+
+    function decreaseAllowance(address spender, uint256 subtractedValue, uint256 period) public virtual returns (bool) {
+        address owner = _msgSender();
+        uint256 currentAllowance = allowance(owner, spender);
+        require(currentAllowance >= subtractedValue, "ERC20: decreased allowance below zero");
+        unchecked {
+            _approve(owner, spender, currentAllowance - subtractedValue, period);
+        }
+
+        return true;
+    }
+```
+
+The above modifications adds a validity period to ERC-20 token approvals. Users can freely update their allowance validity at any time by calling `approve(address spender, uint256 amount, uint256 period)`.
+
+## Rationale
+`_expiration` is implemented as a mapping to improve the compatibility of the code.
+
+So that existing event statistics are not perturbed by the implementation of this standard, `_approve()` is triggered by the new event `AllowanceExpireTimeUpdated()`.
+
+This standard is absolutely compatible with the traditional ERC-20 standard.  `DEFAULT_EXPIRATION` is added so that the original `approve(address spender, uint256 amount)` method header can remain for backwards compatibility and ease of programming. `DEFAULT_EXPIRATION` can be set to a constant or variable according to the developer's needs.
```

Comment:
Please move this paragraph to the Backwards Compatibility section.

Reply:
committed